src/lib/ModelConfig.cc

```diff
@@ -286,40 +286,46 @@ void ModelConfig::validate() {
                         std::vector<bool>{false, true}, ExpectedValue::Bool);
       if (euNode[subFields[1]].IsDefined() &&
           !(euNode[subFields[1]].IsNull())) {
-        // Compile set of blocking groups into a queue
-        std::queue<uint16_t> blockingGroups;
-        for (size_t j = 0; j < euNode[subFields[1]].size(); j++) {
-          char bgNum[50];
-          snprintf(bgNum, 50, "Blocking group %zu", j);
-          if (nodeChecker<std::string>(
-                  configFile_[root][i][subFields[1]][j],
-                  (std::string(euNum) + std::string(bgNum)), groupOptions_,
-                  ExpectedValue::String)) {
-            uint16_t mappedGroup =
-                groupMapping_[euNode[subFields[1]][j].as<std::string>()];
-            blockingGroups.push(mappedGroup);
-            configFile_["Execution-Units"][i]["Blocking-Groups"][j] =
-                mappedGroup;
+        if (configFile_["Core"]["Simulation-Mode"].as<std::string>() ==
+            "inorderpipelined") {
+          invalid_ << "\t - Execution blocking groups are not compatible with "
+                      "an inorderpipelined Simulation Mode\n";
```

Comment:
The flow of instructions in the EU must be inorder because of how we stall at the WB unit (staling the completion slots). With blocking groups, it's possible for uops to flow out of order within the EU and thus a uop gets stuck in the EU behind a younger-decoded uop waiting for that uop to enter the WB. 

A solution would be to instead queue all the instructions in the WB unit (creates issues with an ever-growing queue whilst the next uop to WB is executing which then has to be searched through to find the next program-ordered uop to WB) or to queue it within the EU unit (extra complexity all for the inorder core and has the same problem of search or ordering uops as the first solution). 

Additionally, unlike the OoO core, blocking groups within an in-order context seems weird. You'd be partially stalling entries in the EU for them to fully stalled at the pipeline at WB. You could also say that about the internal pipeline of the EU but it may help with pipeline bubbles when concerned with RAW/WAW dependencies.

Reply:
That all makes sense, thanks for the explanation